src/lib.rs

```diff
@@ -239,10 +243,28 @@ pub struct PayOfferParams {
     /// The path we will send back to the offer creator, so it knows where to send back the
     /// invoice.
     pub reply_path: Option<BlindedPath>,
+    /// The amount of time in seconds that we will wait for offer creator to respond with
+    /// an invoice. If not provided, we will use the default value.
+    pub response_invoice_timeout: Option<u32>,
 }
 
 impl OfferHandler {
-    pub fn new() -> Self {
+    pub fn new_default() -> Self {
+        Self::new(None)
+    }
+
+    pub fn new(response_invoice_timeout: Option<u32>) -> Self {
+        let response_invoice_timeout = match response_invoice_timeout {
+            Some(timeout) => {
+                if timeout < 1 {
```

Comment:
Ah is this to make sure a user doesn't set a timeout of zero? I think it'd be good to do validation of this first -- so before OfferHandler::new is called in main.rs, check that the timeout isn't less than 1. If it is less than one log an error! an exit so the user can then will provide the right number.

Reply:
I addressed this in https://github.com/lndk-org/lndk/pull/148/commits/6f18e3899e75426c4c10b928c95117654e11dcb9